app.rb

```diff
@@ -638,9 +638,31 @@ def render_template(name)
   #
   # @todo danielp 2013-06-26: this should be some sort of discovery, not a
   # hand-coded list, but ... it will do, for now.
-  COLLECTIONS = [:brokers, :repos, :tags, :policies,
-                 [:nodes, {'start' => {"type" => "number"}, 'limit' => {"type" => "number"}}], :tasks, :commands,
-                 [:events, {'start' => {"type" => "number"}, 'limit' => {"type" => "number"}}], :hooks, :config]
+  COLLECTIONS = [
+    :brokers,
+    :repos,
+    :tags,
+    :policies,
+    [:nodes, {'start' => {"type" => "number"}, 'limit' => {"type" => "number"}}],
+    :tasks,
+    :commands,
+    [:events, {'start' => {"type" => "number"}, 'limit' => {"type" => "number"}}],
+    :hooks,
+    :config
+  ].map do |coll|
+    # All collection elements save for :config will have a depth parameter
+    # which has the following semantics:
+    #   depth = 0 will return the list of items in the collection
+    #
+    #   depth = 1 will return a detailed list of items in the collection,
+    #   i.e. for each item, it will return whatever collection/<coll>/<item_id>
+    #   returns.
+    next coll if coll == :config
+    depth_param = { 'depth' => { "type" => "number" } }
```

Comment:
I'm not sure the algorithm here is necessary. I'd lean toward something like this:
```
start = {'start' => {'type' => 'number'}}
limit = ...
depth = ...
COLLECTIONS = [
  [:brokers, depth],
  [:repos, depth],
  ...
  [:nodes, start, limit, depth]
  ...
]
```
That to me makes it simpler regarding what is available for each collection. Variable names for `start`, `limit`, `depth` could be better.